newIDE/app/src/SceneEditor/index.js

```diff
@@ -542,13 +538,21 @@ export default class SceneEditor extends React.Component<Props, State> {
   };
 
   _onInstancesSelected = (instances: Array<gdInitialInstance>) => {
-    this.setState({
-      selectedObjectNames: uniq(
-        instances.map(instance => instance.getObjectName())
-      ),
-    });
-    this.forceUpdatePropertiesEditor();
-    this.updateToolbar();
+    const { project, layout } = this.props;
+    const instancesNames = instances.map(e => e.getObjectName());
+    const selectedObjects = enumerateObjects(
+      project,
+      layout
+    ).allObjectsList.filter(e => instancesNames.includes(e.object.getName()));
```

Comment:
I'm a bit afraid of the efficiency of this computation:
- First you enumerate both the project and the layout objects
- Then, for each object, you search its presence in the instancesNames array
I think you can at least keep the `uniq` method to reduce the cost of the second step.
Also, it might be worth adding the possibility to filter objects by name in the `enumeratObjects` method.
Also, can you give a human readable name to the variable `e`? I thought it was supposed to represent an event at first.